code/modules/hydroponics/hydroponics_reagents.dm

```diff
@@ -0,0 +1,507 @@
+		"fluorine" =        2.5,
+		"chlorine" =        1.5,
+		"sacid" =           1.5,
+		"pacid" =           3,
+		"plantbgone" =      3,
+		"cryoxadone" =     -3,
+		"radium" =          2
+		)
+	var/global/list/nutrient_reagents = list(
+		"milk" =            0.1,
+		"beer" =            0.25,
+		"phosphorus" =      0.1,
+		"sugar" =           0.1,
+		"sodawater" =       0.1,
+		"ammonia" =         1,
+		"diethylamine" =    2,
+		"nutriment" =       1,
+		"adminordrazine" =  1,
+		"eznutrient" =      1,
+		"robustharvest" =   1,
+		"left4zed" =        1
+		)
+	var/global/list/weedkiller_reagents = list(
+		"fluorine" =       4,
+		"chlorine" =       3,
+		"phosphorus" =     2,
+		"sugar" =         -2,
+		"sacid" =          2,
+		"pacid" =          4,
+		"plantbgone" =     8,
+		"adminordrazine" = 5
+		)
+	var/global/list/pestkiller_reagents = list(
+		"sugar" =         -2,
+		"diethylamine" =   2,
+		"adminordrazine" = 5
+		)
+	var/global/list/water_reagents = list(
+		"water" =           1,
+		"adminordrazine" =  1,
+		"milk" =            0.9,
+		"beer" =            0.7,
+		"flourine" =       -0.5,
+		"chlorine" =       -0.5,
+		"phosphorus" =     -0.5,
+		"water" =           1,
+		"sodawater" =       1,
+		)
+	var/global/list/mutagenic_reagents = list(
+		"radium" =  0.6,
+		"mutagen" = 1
+		)
+	var/global/list/aging_reagents = list(
+		"clonexadone" =  -2
+		)
+	var/global/list/growspeed_reagents = list( //TODO
+		"radium" =  0.6,
+		"mutagen" = 1
+		)
+
+	// Beneficial reagents have values for modifying health, yield_mod and mut_mod (in that order).
+	var/global/list/beneficial_reagents = list(
+		"beer" =           list( -0.05, 0,   0   ),
+		"fluorine" =       list( -2,    0,   0   ),
+		"chlorine" =       list( -1,    0,   0   ),
+		"phosphorus" =     list( -0.75, 0,   0   ),
+		"sodawater" =      list(  0.1,  0,   0   ),
+		"sacid" =          list( -1,    0,   0   ),
+		"pacid" =          list( -2,    0,   0   ),
+		"plantbgone" =     list( -2,    0,   0.2 ),
+		"cryoxadone" =     list(  3,    0,   0   ),
+		"clonexadone" =    list(  5,    0,   0   ),
+		"ammonia" =        list(  0.5,  0,   0   ),
+		"diethylamine" =   list(  1,    0,   0   ),
+		"nutriment" =      list(  0.5,  0,   0 	 ),
+		"radium" =         list( -1.5,  0,   0.2 ),
+		"adminordrazine" = list(  1,    1,   1   ),
+		"robustharvest" =  list(  0,    0,   0   ),
+		"left4zed" =       list( -0.1,  0,   2   )
+		)
+
+	//Stat-altering reagents have values for modifying: Endurance, Lifespan, Potency, Yield, Nutrient Consumption, in that order.
+	//The stats listed here are only the base amount, the actual effect they have is later randomized and potentially affected by diminishing returns.
+	var/global/list/stat_altering_reagents = list( //TODO
+		"robustharvest" =  list( -0.4, -0,4,  1, 0, 0) ,
+		"diethylamine"  =  list(  0.4,  0,4,  0, 0, 0)
+		)
+*/
+
+/*
+//Process reagents being input into the tray.
+/obj/machinery/portable_atmospherics/hydroponics/proc/process_reagents()
```

Comment:
This world is unkempt, it is impure. We're all sinners here. We deserve to have this awful code around us, to remind us of how bad everything is.